config/include.go

```diff
@@ -133,6 +225,174 @@ func (targetConfig *TerragruntConfig) Merge(sourceConfig *TerragruntConfig, terr
 	}
 }
 
+// DeepMerge performs a deep merge of the given sourceConfig into the targetConfig. Deep merge is defined as follows:
+// - For simple types, the source overrides the target.
+// - For lists, the two attribute lists are combined together in concatenation.
+// - For maps, the two maps are combined together recursively. That is, if the map keys overlap, then a deep merge is
+//   performed on the map value.
+// - Note that some structs are not deep mergeable due to an implementation detail. This will change in the future. The
+//   following structs have this limitation:
+//     - remote_state
+//     - generate
```

Comment:
Out of curiosity, what is the limitation?

Reply:
I'm going to assume you got the answer from the maintainer's note down below, so will mark as resolved.